src/TomcatSeverTreeProvider.ts

```diff
@@ -39,21 +41,23 @@ export class TomcatSeverTreeProvider implements vscode.TreeDataProvider<TreeItem
             const webapps: string = path.join(server.getStoragePath(), 'webapps');
             const iconPath: string = this._context.asAbsolutePath(path.join('resources', 'war.jpg'));
             if (await fse.pathExists(webapps)) {
-                let wars: string[] = await fse.readdir(webapps);
+                const wars: string[] = [];
                 let temp: fse.Stats;
+                let fileExtension: string;
                 // show war packages with no extension if there is one
                 // and no need to show war packages if its unzipped folder exists
-                wars = wars.filter((w: string) => {
+                const promises: Promise<void>[] = (await fse.readdir(webapps)).map(async (w: string) => {
                     if (w.toUpperCase() !== 'ROOT') {
-                        temp = fse.statSync(path.join(webapps, w));
-                        if (temp.isDirectory() || (temp.isFile() && w.endsWith('.war'))) {
-                            return w;
+                        temp = await fse.stat(path.join(webapps, w));
+                        fileExtension = path.extname(path.join(webapps, w));
+                        if (temp.isDirectory() || (temp.isFile() && fileExtension === Constants.WAR_FILE_EXTENSION)) {
+                            wars.push(fileExtension === Constants.WAR_FILE_EXTENSION ? w.substring(0, w.indexOf(fileExtension)) : w);
```

Comment:
Shall we replace ` w.substring(0, w.indexOf(fileExtension)` with `path.basename(w, fileExtension)`?

Reply:
yes, this looks nice~